applications/GeoMechanicsApplication/custom_strategies/schemes/geomechanics_time_integration_scheme.hpp

```diff
@@ -0,0 +1,349 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#pragma once
+
+#include "solving_strategies/schemes/scheme.h"
+
+namespace Kratos {
+
+template <class TSparseSpace, class TDenseSpace>
+class GeoMechanicsTimeIntegrationScheme : public Scheme<TSparseSpace, TDenseSpace> {
+public:
+    using BaseType              = Scheme<TSparseSpace,TDenseSpace>;
+    using DofsArrayType         = typename BaseType::DofsArrayType;
+    using TSystemMatrixType     = typename BaseType::TSystemMatrixType;
+    using TSystemVectorType     = typename BaseType::TSystemVectorType;
+    using LocalSystemVectorType = typename BaseType::LocalSystemVectorType;
+    using LocalSystemMatrixType = typename BaseType::LocalSystemMatrixType;
+
+    void GetDofList(const Element& rElement,
+                    Element::DofsVectorType& rDofList,
+                    const ProcessInfo& rCurrentProcessInfo) override
+    {
+        const bool isActive = (rElement.IsDefined(ACTIVE)) ? rElement.Is(ACTIVE) : true;
+        if (isActive)
+            rElement.GetDofList(rDofList, rCurrentProcessInfo);
+    }
+
+    void GetDofList(const Condition& rCondition,
+                    Element::DofsVectorType& rDofList,
```

Comment:
There is Condition::DofsVector which is the same as Element::DofsVector. I think it is better to use only Conditions or only Elements. From my viewpoint the best way is to keep DofsVector separately, not in Element nor Condition. What do you think?

Reply:
Good catch, I at least made it consistent (so Condition& also uses Condition::DofsVectorType&), but at some point we should just have a header somewhere that defines these things once and once only.